packages/confirm-dialog/src/vaadin-confirm-dialog.js

```diff
@@ -366,6 +389,19 @@ class ConfirmDialog extends ElementMixin(ThemePropertyMixin(ControllerMixin(Poly
     this.addController(this._confirmController);
   }
 
+  /** @private */
+  __accessibleDescriptionRefChanged(_overlayElement, accessibleDescriptionRef) {
+    if (!_overlayElement || accessibleDescriptionRef === this.__oldAccessibleDescriptionRef) {
```

Comment:
The `accessibleDescriptionRef === this.__oldAccessibleDescriptionRef` check is probably redundant. Polymer doesn't trigger observers when the property value remains the same by reference.

Reply:
This is actually to avoid calling `setAriaIDReference` with `{newId: undefined, oldId: undefined, fromUser: true}` when the observer is called on attach. I changed the condition to make it more clear.